### ISSUE TITLE
Feature request: Collapse/Fold window? 

### ISSUE THREAD

**Issue by tamis-laan** [qtile/qtile#2315]
It would be beneficial if we could collapse or fold windows. I often find myself with a terminal window that runs some automatic build process that I only need to restart or observe once in a while. One visible line of output would be enough feedback to determine if I need to "unfold" the window.

Now in some way this is already possible with qtile using the `lazy.layout.minimize()` function, which simply minimizes the window to be as small as possible. 

However if you start resizing other windows the minimized window will also change in size, it does not stay in a "folded/collapsed" state so to speak.

**Comment by ramnes** [comment]
What about `lazy.window.toggle_minimize()`?

**Comment by tamis-laan** [comment]
> What about `lazy.window.toggle_minimize()`?

Exactly, this is the behavior I'm looking for, however when you toggle this function it first maximizes the window and in the next toggle it wil minimize, causing other windows to reset there size. 

Second, when resizing other windows it wil start to growing out from it's folded state.

It would be cool if the window would minimize or "fold" and stay in this minimized "folded" state :-)

**Comment by ramnes** [comment]
I don't understand what you're talking about. `lazy.window.toggle_minimize()` just hides or unhides the current window, it doesn't maximizes it, nor put it in a state in which it can be resized.

**Comment by tamis-laan** [comment]
Ah sorry I got confused with `lazy.layout.maximize()`, my config is a bit of a mess. 

The `lazy.window.toggle_minimize()` just minimizes the window right. Causing the window to be completely hidden away.
